modules/product/src/main/java/com/opengamma/strata/product/ReferenceSecurity.java

```diff
@@ -0,0 +1,147 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.product;
+
+import java.time.LocalDate;
+import java.util.Optional;
+
+import com.google.common.collect.ImmutableSet;
+import com.opengamma.strata.basics.currency.Currency;
+import com.opengamma.strata.basics.market.ReferenceData;
+import com.opengamma.strata.collect.Messages;
+
+/**
+ * A single security that can be traded.
+ * <p>
+ * A security is one of the building blocks of finance, representing a fungible instrument that can be traded.
+ * This is intended to cover instruments such as listed equities and futures.
+ * It is intended that Over-The-Counter (OTC) instruments, such as an interest rate swap,
+ * are embedded directly within the trade, rather than handled as one-off securities.
+ * <p>
+ * A security may be referred to using a {@link SecurityId}.
+ * The identifier can be looked up in {@link ReferenceData}.
+ * <p>
+ * Implementations of this interface must be immutable beans.
+ */
+public interface ReferenceSecurity {
+
+  /**
+   * Gets the security identifier.
+   * <p>
+   * This identifier uniquely identifies the security within the system.
+   * It is the key used to lookup the security in {@link ReferenceData}.
+   * <p>
+   * A real-world security will typically have multiple identifiers.
+   * The only restriction placed on the identifier is that it is sufficiently
+   * unique for the reference data lookup. As such, it is acceptable to use
+   * an identifier from a well-known global or vendor symbology.
+   * 
+   * @return the security identifier
+   */
+  public abstract SecurityId getSecurityId();
+
+  /**
+   * Gets the currency that the security is traded in.
+   * 
+   * @return the trading currency
+   */
+  public abstract Currency getCurrency();
+
+  /**
+   * Gets the underlying security identifiers.
+   * <p>
+   * The set must contain all the security identifiers that this security refers to.
```

Comment:
Some more detail would be good about what "underlying" means here, including an example.